multi-system/flake.nix

```
@@ -0,0 +1,50 @@
+{
+  description = "A very basic flake";
+
+  inputs.nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+
+  outputs =
+    { self, nixpkgs }:
+    let
+      # Systems supported by this flake
+      supportedSystems = [
+        "x86_64-linux"
+        "aarch64-linux"
+        "x86_64-darwin"
+        "aarch64-darwin"
+      ];
+      # A helper function for specifying per-system outputs
+      forEachSupportedSystem =
+        f:
+        nixpkgs.lib.genAttrs supportedSystems (
+          system:
+          f {
+            pkgs = import nixpkgs { inherit system; };
+          }
+        );
```

Is this really idiomatic? I find myself seeing this far more:
```suggestion
      forAllSystems =
        f:
        nixpkgs.lib.genAttrs supportedSystems (
          system:
          f nixpkgs.legacyPackages.${system}
        );
```

I feel that in the context of a template, it's important to match idioms and prevent splintering.

---

I don't think the `legacyPackages` way should be idiomatic because it doesn't allow you to pass in configuration:

```nix
{
  pkgs = import nixpkgs {
    inherit system;
    config.allowUnfree = true;
  };
}
```